src/liboslcomp/codegen.cpp

```diff
@@ -648,27 +648,27 @@ ASTNode::one_default_literal (const Symbol *sym, ASTNode *init,
         completed = false;
     } else if (type.is_int()) {
         if (islit && lit->typespec().is_int())
-            out += Strutil::format ("%d", lit->intval());
+            out += Strutil::sprintf ("%d", lit->intval());
         else {
             out += "0";  // FIXME?
             completed = false;
         }
     } else if (type.is_float()) {
         if (islit && lit->typespec().is_int())
-            out += Strutil::format ("%d", lit->intval());
+            out += Strutil::sprintf ("%d", lit->intval());
         else if (islit && lit->typespec().is_float())
-            out += Strutil::format ("%.8g", lit->floatval());
+            out += Strutil::sprintf ("%.8g", lit->floatval());
```

Comment:
Not related to this refactor obviously, but why are these `%.8g` and not `%.9g` ?

Reply:
I can only guess that at the time I thought 8 was plenty? I'll fix that while I'm at it.

Reply:
%.9g is the minimum that's guaranteed to be full precision for all fp32 code points?

Reply:
Actually, there are number of %.8g instances in the code base. I think I'll make that fix a separate PR to isolate it.

Reply:
Yes.

https://randomascii.wordpress.com/2013/02/07/float-precision-revisited-nine-digit-float-portability/

Reply:
I'm not going to fix the precision for this patch, that will be the next one.